project.py

```diff
@@ -0,0 +1,174 @@
+        product = ' '.join(speech)
+    else:
+        product = ' '.join(speech) + '.'
+    product = product.replace('  ', ' ')        #replace all double spaces with single spaces.
+    return product
+
+if __name__ == '__main__':
+    max_season = 3
+    allscripts = ''
+    my_file = Path("dict1.p")
+    my_file2 = Path("dict2.p")
+    my_file3 = Path("allwords.p")
+    remake = False
+    y = input("Do you want to customize the number of seasons? (y/n)")
+    if y.lower() == 'y':
+        max_seasons = input("How many seasons do you want?: ")
+        remake = True
+    # print(not my_file.is_file()) #For testing purposes
+    # print(not my_file2.is_file()) #For testing purposes
+    # print(not my_file3.is_file()) #For testing purposes
+    if (not my_file.is_file()) or (not my_file2.is_file()) or (not my_file3.is_file()) or remake: #check to see if there are already dictionaries; if there aren't then dictionaries are created.
+        basehtml = 'http://www.springfieldspringfield.co.uk/view_episode_scripts.php?tv-show=family-guy&episode=s'
+        temptml = basehtml
+        episodes_in_season = [-1, 8, 22, 23, 28, 19, 13, 17, 22, 19, 24, 23, 22, 19, 21, 14]
+        for season in range (1, max_season):
+            for i in range (1,episodes_in_season[season]):   #rewritten for all seasons; writes the link in a way to grab the seasons correctly
+                if season < 10:
+                    if i < 10:
+                        temp =  temptml + '0%de' % season + '0%d' % (i)
+                    else:
+                        temp =  temptml + '0%de' % season + '%d' % (i)
+                else:
+                    if i < 10:
+                        temp =  temptml + '%de' % season + '0%d' % (i)
+                    else:
+                        temp =  temptml + '%de' % season + '%d' % (i)
+                html = BeautifulSoup(requests.get(temp).text, 'lxml')
+                thing = html.find("div", class_ = 'scrolling-script-container')
+                allscripts = allscripts + " " + str(thing.text)
+
+        #gather the trump speeches
+        html = BeautifulSoup(requests.get('https://github.com/PedramNavid/trump_speeches/blob/master/data/full_speech.txt').text, 'lxml')
+        thing = html.find("table", class_ = 'highlight tab-size js-file-line-container')
+        allscripts += thing.text
+
+        #pickle the necessary information
+        pickle.dump(allscripts, open("allwords.p", "wb"))
+        pickle.dump(dict_creation_one(allscripts), open("dict1.p", "wb"))
+        pickle.dump(dict_creation_two(allscripts), open("dict2.p", "wb"))
+        pickle.dump(allscripts, open("backup.p", "wb"))
+
+    #call the dictionaries/all words to run the program with
+    oneworddict = pickle.load( open( "dict1.p", "rb"))
+    twoworddict = pickle.load( open("dict2.p", "rb"))
+    allwords = pickle.load( open("allwords.p", "rb"))
+    blah2 = markov_chain_two(twoworddict, oneworddict, allwords) #create the markov chain
```

Comment:
Not digging the variable choice, but the comments do help clarify what the code's doing in this section and all.